xtext-website/_posts/releasenotes/2019-06-04-version-2-18-0.md

```diff
@@ -6,10 +6,96 @@ categories: releasenotes
 published: false
 ---
 
-Bla
+Xtext 2.18.0 is mainly a maintenance release.
 
-## Bla
-Bla
+## Improved Java 11 Support
+
+Xtext's support for Java 11 has been further improved:
+
+* Improved annotation processing in Xbase. ([xtext-extras#425](https://github.com/eclipse/xtext-extras/issues/425), [xtext-xtend#800](https://github.com/eclipse/xtext-xtend/issues/800))
+* Several features are relying on querying the classpath, e.g. content assist. ([xtext-web#128](https://github.com/eclipse/xtext-web/issues/128))
+* New Java 11 API methods broke test assumptions when running on a Java 11 VM. All affected test cases have been refactored. ([xtext-extras#424](https://github.com/eclipse/xtext-extras/issues/424))
+
+## Eclipse Integration
+
+### Search References Result Page: New Context Actions
+
+The results of the _Search References_ action can now be copied to the clipboard. Further an action has been added to remove search results from the view. Both actions are available from the context menu and by common keyboard shortcuts.
+
+![Search References Result Page: New Context Actions]({{site.baseurl}}/images/releasenotes/2_18_SearchReferencesResultActions.png){:height="50%" width="50%"}
+
+
+### New Quickfix on Empty Keywords
+
+Grammar rules with empty keyword statements (`''`) are raising errors already. Two new quickfixes have been added that can be used to resolve these errors:
+
+* Remove the empty keyword
+* Set the keyword value from the declaring rule name
+
+([xtext-eclipse#994](https://github.com/eclipse/xtext-eclipse/issues/994))
+
+### Fixed Potential Deadlock on Startup
+
+A deadlock may occur on startup of Eclipse. The builder infrastructure has been improved to avoid locking on loading its stored state. ([#1460](https://github.com/eclipse/xtext/issues/1460))
+
+### Fixed: Exceptions for Closed Projects
+
+When projects are removed from the workspace the `ProjectOpenedOrClosedListener` may caused exceptions that were logged to the error log. This situation has been fixed with [xtext-eclipse#1066](https://github.com/eclipse/xtext-eclipse/issues/1066).
+
+### Refactoring Support: Improved Progress Monitoring
```

Comment:
reporting?

Reply:
"reporting" is better here, although cancellation is not really reporting